### PR TITLE
[TE] Created a separate doc for Application. +cosmetic refactor

### PR DESCRIPTION
**Pull request by suvodeep-pyne**
## Description
Refactored ThirdEye Documentation. 
- Created a separate document for Application.
- Moved content from Configuration and Getting Started doc into this doc.
- Added documentation to create TE application via API

### CODE REVIEW

File: thirdeye/docs/application.rst [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+2. Create the application using the command below. Feel free to update the inline json as per your needs.
+
+.. code-block:: bash
+
+  function create_te_app {
+  	token=$1
+  	curl --location --request POST 'http://localhost:1426/thirdeye/entity?entityType=APPLICATION' \
+  		--header "Authorization: Bearer ${token}" \
+  		--header 'Content-Type: application/json' \
+  		--header 'Cookie: te_auth=8ba22acc7616b77fb62fc0aad88a638a16c74e7922209d30afe8a96fee4b55e9' \
```
Comment (akshayrai) [inline thirdeye/docs/application.rst:89]: Shouldn't the te_auth token be replaced here?

File: thirdeye/docs/application.rst [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+An application is a basic TE entity can serves as a container for metrics, alerts and other entities.
+It also can be used to group a bunch of users.
+
+Creating an Application
+##########################################
+
+You can create an Application in ThirdEye using 2 ways:
+1. ThirdEye Admin
+2. Using the API
```
Comment (akshayrai) [inline thirdeye/docs/application.rst:34]: Right now, the admin portal doesn't have the capability to add a new application for the first time. We need to create an application using the API first and subsequently, we can create more applications from the portal. This is a minor UI bug.

Issue: https://github.com/apache/incubator-pinot/issues/5525

Comment (suvodeep-pyne) [reply thirdeye/docs/application.rst:34]: Hey @akshayrai I'm able to create an application from the UI. In fact, I was documenting this while I was going through the process.

File: thirdeye/docs/application.rst (hunk 1 above)
Comment (suvodeep-pyne) [reply thirdeye/docs/application.rst:89]: Yes. Thanks for pointing that out. I'll update the PR.

File: thirdeye/docs/application.rst (hunk 2 above)
Comment (akshayrai) [reply thirdeye/docs/application.rst:34]: Awesome! Looks like it was added recently: https://github.com/apache/incubator-pinot/pull/5601

**Pull request merged**